testkit/src/main/scala/org/bitcoins/testkit/Implicits.scala

```diff
@@ -0,0 +1,23 @@
+package org.bitcoins.testkit
+
+import org.scalacheck.Gen
+import scala.annotation.tailrec
+
+/**
+  * Provides extension methods, syntax
+  * and other handy implicit values that
+  * aid in testing.
+  */
+object Implicits {
+
+  /** Extension methods for Scalacheck generatos */
+  implicit class GeneratorOps[T](private val gen: Gen[T]) extends AnyVal {
+
+    @tailrec
+    /** Gets a sample from this generator that's not `None` */
+    def sampleSome: T = gen.sample match {
```

Comment:
This is meant as a better alternative than `FooGenerator.sample.get`. In many tests in the code base we don't want to have a full blown property-based test (for example while doing wallet tests, running the same test 10 times with marginally different inputs would slow things down _a lot_), but we need testing data that is actually meaningful (like for example a transaction paying to a specific SPK)

Reply:
This can also slow down the code significantly because it is guaranteed that we generate a value. This can essentially end up in an infinite loop trying to satisfy the generator. That is why the `Option` type is returned.

Reply:
I think we should just cache these values in a `*TestUtil.scala` file if you don't want to do the full blow property based testing.

Reply:
If we have a generator that leads to a significant slowdown/infinite loop I'd say that's a problem on its own, as that's going to make any tests using that generator equally slow. The solution in that case is fixing the generator. 

For a bunch of things, having a `"TestUtil` object works fine, but what about for example an address that pays to a specific SPK, and that SPK is generated by the wallet in a test? I'm thinking `sampleSome` works fine in that case. 

If you'd like, I could add a constraint to the method that it's recursing maximum 10 levels deep, is that something you could live with?

Reply:
If you feel strongly about it I will let it go, but I think this is a classic example of circumventing the type system safety to be more convenient. Circumventing the type system is more acceptable in test code, but I do think it will bite us in the long term since the library designer designed the API like that for a reason.